### ISSUE TITLE
z:acceptance governance fails/flakes: No quorum

### ISSUE THREAD

**Issue by dckc** [Agoric/Agoric#10708]
I'm getting a ci failure in an unrelated PR:
https://github.com/Agoric/agoric-sdk/pull/10692#issuecomment-2547042803

This `readLatestHead` immediate assumes vstorage updates by the time the deadline is reached; I don't think that's necessarily the case:

https://github.com/Agoric/agoric-sdk/blob/3e27c74e5385180799f500232d795508e275c687/a3p-integration/proposals/z%3Aacceptance/governance.test.js#L100-L106

A `retryUntilCondition` seems in order.

I don't know whether it's a flake. It failed again.

https://github.com/Agoric/agoric-sdk/actions/runs/12345315961/job/34449230334?pr=10692#step:10:8388

cc @frazarshad @LuqiPan  @Chris-Hibbert

**Comment by dckc** [comment]
On second thought, I wonder if `retryUntilCondition` is really going to help: we get `no quorum`, not an empty read.

But the `no quorum` could be the outcome of an earlier question.

Be sure to check that `latestQuestion` has updated to the relevant question before checking `latestOutcome`.

**Comment by dckc** [comment]
possibly out of scope, but while you're in there... I'd like the previousOfferId arg of `governanceDriver.proposeVaultDirectorParamChange` become optional; the function should do the `.find` in the wallet purses and such.

**Comment by anilhelvaci** [comment]
## Investigation
### Param Change Tx
```
2024-12-16T04:10:17.755Z SwingSet: vat: v43: walletFactory.fromBridge: { blockHeight: 1765, blockTime: 1734322212, owner: 'agoric1c9gyu460lu70rtcdp95vummd6032psmpdx7wdy', 
spendAction: '{
"body":"#{\\"method\\":\\"executeOffer\\",\\"offer\\":{\\"id\\":\\"propose-1734322211878\\",\\"invitationSpec\\":{\\"invitationMakerName\\":\\"VoteOnParamChange\\",\\"previousOffer\\":\\"charter-1734320369849\\",\\"source\\":\\"continuing\\"},
\\"offerArgs\\":{\\"deadline\\":\\"+1734322222\\",\\"instance\\":\\"$0.Alleged: BoardRemoteInstanceHandle\\",\\"params\\":{\\"ChargingPeriod\\":\\"+400\\"},\\"path\\":{\\"paramPath\\":{\\"key\\":\\"governedParams\\"}}
},\\"proposal\\":{}}}","slots":["board00360"]}', type: 'WALLET_SPEND_ACTION' }
```

### Voting Duration
https://github.com/Agoric/agoric-sdk/blob/9e124c3c646887571c0d050e692c09b8eb061aeb/a3p-integration/proposals/z%3Aacceptance/test-lib/governance.js#L147

### Thoughts
The `deadline`'s GMT conversion is `Mon Dec 16 2024 04:10:22 GMT+0000`. Given that the voting durations is 10 seconds, the transaction is sent at prepared and sent at `Mon Dec 16 2024 04:10:11 GMT+0000` (double checked with the offerId `propose-1734322211878`). However, when we look at the block timestamp of the governance transaction: `2024-12-16T04:10:17.755Z`, it is clear that there's a delay around 5 seconds.

We can verify the delay by looking at when the `No Quorum` is printed:

```
2024-12-16T04:10:28.658Z SwingSet: xsnap: v43: UnhandledPromiseRejectionWarning: No quorum
2024-12-16T04:10:28.661Z SwingSet: xsnap: v82: UnhandledPromiseRejectionWarning: No quorum
2024-12-16T04:10:28.663Z SwingSet: xsnap: v9: UnhandledPromiseRejectionWarning: No quorum
```

The `deadline` should be at around `2024-12-16T04:10:22` however it is actually `2024-12-16T04:10:28`. So the 5 secondish delay is now ceils up to 6 seconds.

And we can also see that the Econ Committee members' votes are getting into block (`2024-12-16T04:10:32`) AFTER the quorum is reached:

```
2024-12-16T04:10:32.772Z SwingSet: vat: v43: walletFactory.fromBridge: { blockHeight: 1768, blockTime: 1734322227, owner: 'agoric1wrfh296eu2z34p6pah7q04jjuyj3mxu9v98277', 
spendAction: '{"body":"#{\\"method\\":\\"executeOffer\\",\\"offer\\":{\\"id\\":\\"propose-1734322229372\\",
\\"invitationSpec\\":{\\"invitationArgs\\":[[{\\"changes\\":{\\"ChargingPeriod\\":\\"+400\\"}}],\\"$0.Alleged: BoardRemoteQuestionHandle\\"],\\"invitationMakerName\\":\\"makeVoteInvitation\\",
\\"previousOffer\\":\\"gov-committee-1734320362167\\",\\"source\\":\\"continuing\\"},\\"proposal\\":{}}}","slots":["board061141"]}', type: 'WALLET_SPEND_ACTION' }
```

This whole situation, I believe, is due to a conjunction with the auction's `clockStep`. When there is a clockStep, the transactions are delayed to later blocks to be processed. Since the `votingDuration` is only 10 seconds and these delays take around 5-6 seconds. we end up with an election flow that is out of sync. 

## Solution
Increase the `votingDuration`. Start with 20 seconds.

**Comment by Jorge-Lopes** [comment]
The issues here described are addressed in the [PR #10555](https://github.com/Agoric/agoric-sdk/pull/10555)
